internal/postgres/users_test.go

```diff
@@ -192,3 +197,52 @@ COMMIT;`))
 		assert.Equal(t, calls, 1)
 	})
 }
+
+func TestWriteUsersSchemasInPostgreSQL(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("Mixed users", func(t *testing.T) {
+		calls := 0
+		exec := func(
+			_ context.Context, stdin io.Reader, _, _ io.Writer, command ...string,
+		) error {
+			calls++
+
+			b, err := io.ReadAll(stdin)
+			assert.NilError(t, err)
+
+			// The command strings will contain either of two possibilities, depending on the user called.
+			commands := strings.Join(command, ",")
+			re := regexp.MustCompile("--set=databases=\\[\"db1\"\\],--set=username=user-single-db|--set=databases=\\[\"db1\",\"db2\"\\],--set=username=user-multi-db")
+			assert.Assert(t, cmp.Regexp(re, commands))
```

Comment:
👍🏻 Interesting to see that the `postgres.Exec*` design makes it hard to test variables.